File: restaurant/state/cooking_state.py
from state.dish_state import DishState

class CookingState(DishState):
    """Trạng thái: Đơn hàng đang được nấu"""
    
    def handle(self, context):
        """Xử lý khi đơn hàng bắt đầu được nấu"""
        print(f"Đơn hàng {context.order_id} đang được nấu.")
        context.state = CookingState()
    
    def get_status_name(self) -> str:
        return "Cooking"

File: backend/src/main/java/com/restaurant/pattern/state/CookingState.java
package com.restaurant.pattern.state;

import com.restaurant.model.entity.OrderItem;
import com.restaurant.model.enums.DishStatus;

/**
 * Trạng thái: Món đang được nấu
 * Trạng thái tiếp theo: READY
 */
public class CookingState implements DishState {
    
    @Override
    public void nextState(OrderItem orderItem) {
        // Chuyển sang trạng thái READY
        orderItem.setDishStatus(DishStatus.READY);
        orderItem.setState(new ReadyState());
        System.out.println("Món '" + orderItem.getMenuItem().getName()
                + "' đã chuyển sang trạng thái: READY");
    }

    @Override
    public String getStateName() {
        return "COOKING";
    }
}
